sdk/java/pulumi/src/test/java/com/pulumi/automation/LocalWorkspaceTest.java

```diff
@@ -166,13 +642,299 @@ void testStackLifecycleLocalProgram(@EnvVars Map<String, String> envVars) {
+                .description("non-standard description")
+                .build();
+
+        assertThrows(ProjectSettingsConflictException.class, () -> {
+            LocalWorkspace.createStack(projectName, stackName, program, LocalWorkspaceOptions.builder()
+                    .workDir(workDir)
+                    .environmentVariables(env)
+                    .projectSettings(projectSettings)
+                    .build());
+        });
+    }
+
+    @Test
+    void testStateDelete(@EnvVars Map<String, String> envVars) throws Exception {
+        var env = new HashMap<String, String>(envVars);
+        env.put("PULUMI_CONFIG_PASSPHRASE", "test");
+
+        final var type = "test:res";
+        Consumer<Context> program = ctx -> {
+            new ComponentResource(type, "a");
+        };
+
+        var stackName = randomStackName();
+        var projectName = "test_state_delete";
+
+        try (var stack = LocalWorkspace.createStack(projectName, stackName, program, LocalWorkspaceOptions.builder()
+                .environmentVariables(env)
+                .build())) {
+
+            try {
+                // pulumi up
+                var upResult = stack.up();
+                assertThat(upResult.summary().kind()).isEqualTo(UpdateKind.UPDATE);
+                assertThat(upResult.summary().result()).isEqualTo(UpdateState.SUCCEEDED);
+                assertThat(upResult.summary().resourceChanges().get(OperationType.CREATE)).isEqualTo(2);
+
+                // export state
+                var exportResult = stack.exportStack();
+                assertThat(exportResult.deployment()).hasEntrySatisfying("resources", value -> {
+                    assertThat(value).isInstanceOf(List.class);
+                    var resourcesList = (List<?>) value;
+                    assertThat(resourcesList).hasSize(2);
+                });
+                var resources = (List<?>) exportResult.deployment().get("resources");
+                var urn = resources.stream()
+                    .filter(item -> item instanceof Map)
+                    .map(item -> (Map<?, ?>) item)
+                    .filter(itemMap -> itemMap.containsKey("urn") &&
+                                    itemMap.get("urn") instanceof String &&
+                                    ((String) itemMap.get("urn")).contains(type))
+                    .map(itemMap -> (String) itemMap.get("urn"))
+                    .findFirst()
+                    .orElseThrow();
+
+                // pulumi state delete
+                stack.state().delete(urn);
+
+                // test
+                exportResult = stack.exportStack();
+                assertThat(exportResult.deployment()).hasEntrySatisfying("resources", value -> {
+                    assertThat(value).isInstanceOf(List.class);
+                    var resourcesList = (List<?>) value;
+                    assertThat(resourcesList).hasSize(1);
+                });
+            }
+            finally {
```

Comment:
A bit of inconsistent formatting here -- shame Java doesn't have CLI formatters :sob: